### ISSUE TITLE
"Fix whitespace formatting." is not very useful. Please add an option to show a diff of the formatting rule violation.

### ISSUE THREAD

**Issue by siegfriedpammer** [dotnet/format#789]
I have enabled `dotnet-format --check` on our build server, but it fails hard. This is not the case locally, where it runs successfully.

For reference see https://ci.appveyor.com/project/icsharpcode/ilspy/builds/34921427/job/f80qltt0knx4dwka

I use the `--verbosity diagnostic` setting, but the error message "Fix whitespace formatting." is not very useful. How do other people using dotnet-format debug formatting issues on their CI system?

Would it be possible to add a diff of the change that was performed by the formatter? Could also be a new option `--verbosity show-diff` or similar.

Thank you very much!

**Comment by siegfriedpammer** [comment]
I have fixed my build server script. The problem was that appveyor does not use autocrlf=true, however, I would appreciate if the dotnet-format would tell me about these things in the log, without having to RDP into the build server and look for the problem myself. Thanks!

**Comment by JoeRobich** [comment]
Interesting, I would have expected end of line changes to be caught by the EndOfLineFormatter. @sharwell did the Roslyn formatter add end of line support?

**Comment by JoeRobich** [comment]
Although the formatter applies all rules in one pass and is unable to provide information about which code style options resulted in which changes, we could process the TextChanges and generate a more detailed message.
![image](https://user-images.githubusercontent.com/611219/106961601-ad448600-66f2-11eb-9e08-f14c98da8e4f.png)
Would this extra information be useful?

**Comment by siegfriedpammer** [comment]
Yes, I think that would be a very good improvement, however, I would have to try it out and see if it helps solve these issues faster.
